pkg/csi/service/vanilla/controller.go

```diff
@@ -1015,8 +1026,25 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 					"failed to get VolumeID from volumeMigrationService for volumePath: %q", volumePath)
 			}
 		}
+		// Fetch vCenterHost, vCenterManager & volumeManager for given volume, based on VC configuration
+		vCenterHost, err = getVCenterHostForVolumeID(ctx, c, req.VolumeId, volumeInfoService)
+		if err != nil {
+			return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+				"failed to get vCenter Host for volume Id: %q. Error: %v", req.VolumeId, err)
+		}
```

Comment:
Can you Info log the `vCenterHost` you received here as it will be easy for debugging DeleteVolume failures.

Reply:
updated code